src/test/scala/io/iohk/ethereum/jsonrpc/EthProofServiceSpec.scala

```diff
@@ -126,18 +78,113 @@ class EthProofServiceSpec
   }
 
   "EthProofService" should "return an error when the proof is requested for non-existing account" in new TestSetup {
-    val ethGetProof = new EthProofService(blockchain, blockGenerator, blockchainConfig.ethCompatibleStorage)
-    val key = 999
-    val storageKeys = Seq(StorageProofKey(key))
-    val blockNumber = BlockParam.Latest
     val wrongAddress = Address(666)
     val request = GetProofRequest(wrongAddress, storageKeys, blockNumber)
     val retrievedAccountProofWrong: ServiceResponse[ProofService.GetProofResponse] = ethGetProof.getProof(request)
-    retrievedAccountProofWrong.runSyncUnsafe().isLeft shouldBe true
+    val result = retrievedAccountProofWrong.runSyncUnsafe()
+
+    result.isLeft shouldBe true
+    result.fold(l => l.message should include("No account found for Address"), r => r)
+  }
+
+  "EthProofService" should "return the proof with empty value for non-existing storage key" in new TestSetup {
+    val wrongStorageKey = Seq(StorageProofKey(321))
+    val request = GetProofRequest(address, wrongStorageKey, blockNumber)
+    val retrievedAccountProofWrong: ServiceResponse[ProofService.GetProofResponse] = ethGetProof.getProof(request)
+    val result = retrievedAccountProofWrong.runSyncUnsafe()
+    result.isRight shouldBe true
+    result.fold(l => l, r => r.proofAccount.storageProof.map(v => {
+      v.proof.nonEmpty shouldBe true
+      v.value shouldBe BigInt(0)
+    }))
+  }
+
+  "EthProofService" should "return the proof and value for existing storage key" in new TestSetup {
+    val storageKey = Seq(StorageProofKey(key))
```

Comment:
looks like these can be extracted as a common helper (repeated in all tests). maybe:
```scala
val result = fetchProof(address, key, blockNo).runSyncUnsafe()
```